lib/openstudio/lib/openstudio_aws_wrapper.rb

```diff
@@ -118,15 +118,43 @@ def describe_availability_zones_json
     describe_availability_zones.to_json
   end
 
-  def describe_total_instances
+  def total_instances_count
     resp = @aws.describe_instance_status
 
     region = resp.instance_statuses.length > 0 ? resp.instance_statuses.first.availability_zone : 'no_instances'
     { total_instances: resp.instance_statuses.length, region: region }
   end
 
-  def describe_total_instances_json
-    describe_total_instances.to_json
+  # describe the instances by group id
+  def describe_instances
+    resp = nil
+    if group_uuid
+      resp = @aws.describe_instances(
+        filters: [
+          # {name: 'instance-state-code', values: [0.to_s, 16.to_s]}, # running or pending -- any state
+          { name: 'tag-key', values: ['GroupUUID'] },
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.